backend/spec/system_helper.rb

```diff
@@ -0,0 +1,45 @@
+require "capybara/cuprite"
+require "rails_helper"
+
+Capybara.default_max_wait_time = 2
+Capybara.default_normalize_ws = true
+
+Capybara.register_driver(:cuprite) do |app|
+  Capybara::Cuprite::Driver.new(
+    app,
+    window_size: [1200, 800],
+    browser_options: {
+      "no-sandbox": nil
+    },
+    # Increase Chrome startup wait time (required for stable CI builds)
+    process_timeout: 10,
+    inspector: true,
+    headless: ENV["HEADLESS"] != "false"
+  )
+end
+
+# Configure Capybara to use :cuprite driver by default
+Capybara.default_driver = Capybara.javascript_driver = :cuprite
+
+RSpec.configure do |config|
+  config.include CupriteHelpers, type: :system
+  config.include PageHelpers, type: :system
+
+  config.around(:each, type: :system) do |ex|
+    was_host = Rails.application.default_url_options[:host]
+    Rails.application.default_url_options[:host] = Capybara.server_host
+    ex.run
+    Rails.application.default_url_options[:host] = was_host
+  end
+
+  config.prepend_before(:each, type: :system) do
+    driven_by Capybara.javascript_driver
+  end
+
+  config.before(:suite) do
+    Rails.application.load_tasks
+    `yarn build`
+    `yarn build:css`
+    Rake::Task["assets:precompile"].execute
```

Comment:
hmm, didn't this slow down tests a lot. I often run just 1-2 tests during development, but I run them a lot so every second which I spend waiting for tests to finish count. Would leave it to pipeline to run this, but I understand that you wanna ensure that FE is up to date before running tests

Reply:
Yeah, I think that was necessary, took it from other project. I want to avoid the pain :D.

Reply:
I don't actually remember what was the problem, but I will try to check if I left any clues in at least commit messages.